core/src/main/scala/com.snowplowanalytics.stream/loader/transformers/PlainJsonTransformer.scala

```diff
@@ -75,8 +73,11 @@ class PlainJsonTransformer(documentIndex: String, documentType: String)
    */
   private def toJsonRecord(jsonString: String): ValidationNel[String, JsonRecord] = {
     parseOpt(jsonString) match {
-      case Some(jvalue) => JsonRecord(jsonString, None).success
-      case None         => "Json parsing error".failureNel
+      case Some(jvalue) =>
+        JsonRecord(
+          jvalue.asInstanceOf[JObject] ~ ("id" -> UUID.randomUUID().toString),
```

Comment:
asInstanceOf

Reply:
Json4s is a nice library but its inheritance kills! changed.